lib/autocomplete-manager.coffee

```diff
@@ -272,18 +180,166 @@ class AutocompleteManager
     @editor.insertText(match.word)
     @editor.setSelectedBufferRanges(newSelectedBufferRanges)
 
-  # Private: Sets the current buffer, starts listening to change events and delegates
-  # them to #onChanged()
+  # Public: Finds and returns the content before the current cursor position
+  #
+  # selection - The {Selection} for the current cursor position
   #
-  # currentBuffer - The current {TextBuffer}
-  setCurrentBuffer: (@currentBuffer) ->
-    @compositeDisposable.add @currentBuffer.onDidSave(@editorSaved)
-    @compositeDisposable.add @currentBuffer.onDidChange(@editorChanged)
+  # Returns {String} with the prefix of the {Selection}
+  prefixOfSelection: (selection) ->
+    selectionRange = selection.getBufferRange()
+    lineRange = [[selectionRange.start.row, 0], [selectionRange.end.row, @editor.lineTextForBufferRow(selectionRange.end.row).length]]
+    prefix = ''
+    wordRegex = /\b\w*[a-zA-Z_-]+\w*\b/g
+    @editor.getBuffer().scanInRange wordRegex, lineRange, ({match, range, stop}) ->
+      stop() if range.start.isGreaterThan(selectionRange.end)
+
+      if range.intersectsWith(selectionRange)
+        prefixOffset = selectionRange.start.column - range.start.column
+        prefix = match[0][0...prefixOffset] if range.start.isLessThan(selectionRange.start)
+
+    return prefix
+
+  # Private: Checks whether the current file is blacklisted.
+  #
+  # Returns {Boolean} that defines whether the current file is blacklisted
+  currentFileBlacklisted: ->
+    blacklist = (atom.config.get('autocomplete-plus.fileBlacklist') or '')
```

Comment:
You should be able to use a config schema to type this key as an array, avoiding this string manipulation.